include/revng/Pipes/PipelineManager.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <string>
+#include <vector>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/SavableObject.h"
+#include "revng/Pipes/LLVMContextWrapper.h"
+#include "revng/Pipes/ModelGlobal.h"
+
+namespace revng::pipes {
+
+/// This is a "God Object" that can be used when there will be exactly one
+/// Pipeline spawned by this process.
+///
+/// It will care of all configurations including llvm ones and provide a runner
+/// in a usable state.
+class PipelineManager {
+private:
+  explicit PipelineManager() = default;
+
+  std::string ExecutionDirectory;
+  std::unique_ptr<LLVMContextWrapper> Context;
+  std::unique_ptr<ModelGlobal> ModelWrapper;
+  std::unique_ptr<pipeline::Context> PipelineContext;
+  std::unique_ptr<pipeline::Loader> Loader;
+  std::unique_ptr<pipeline::Runner> Pipeline;
+  pipeline::Runner::State State;
```

Comment:
it's just the place where we throw the full state enumeration requested by the c api. the only purpose is to take ownership of the memory so that the user of the c api does not have to free it themself.

Reply:
OK, `LastState`, `CachedState`? State is too general and misleading.